vis_gen.py

```diff
@@ -0,0 +1,565 @@
+r"""Visual (image) generation module for confusable detection."""
+
+import os
+import random
+import shutil
+import argparse
+from argparse import RawDescriptionHelpFormatter
+
+import qahirah as qah
+from qahirah import CAIRO, Colour, Vector
+
+class VisGen:
+    """An character image generator for a specific font face. Also serve as
+    dataset generator.
+    
+    Initialization:
+        >>> vg = VisGen(font_size=28, image_size=36, \
+                        font_name="Noto Sans CJK SC", out_dir="test_out")
+    Configurations:
+        >>> vg.font_name = "Noto Serif CJK SC"
+        >>> vg.font_size = 24
+        >>> vg.font_style = "SemiBold"
+        >>> vg.antialias = "Best"
+        >>> vg.grayscale = True
+    Visualize single character:
+        >>> vg.visualize_single('\u6400')
+    Visualize code point in specific range:
+        >>> vg.visualize_range(start='\u6400', end='\u64ff')
+    Generate dataset from file:
+        >>> vg.generate_dataset_from_file('source/dataset.txt', \
+        >>>     font_styles=['Bold', 'Regular'], \
+        >>>     antialiases=['None', 'Default'])
+    Split into train and test dataset:
+        >>> vg.train_test_split(num_test=200)
+    """
+
+    def __init__(self, font_size=16, image_size=20,
+                 font_name="Noto Sans CJK SC", font_style=None,
+                 antialias="Default", out_dir="img_out", grayscale=False):
+        """Store info about font_size, image_size, out_dir. Search and find
+        specified font_name. Set font weight (Regular, Bold, Semi-bold ...).
+        Set font anti-aliasing style (Best, Default, Fast...). Set output type.
+
+        Args:
+            font_size: Int, size of the font
+            image_size: Int, height and width of the output image (in pixel)
+            font_name: Str, name of the font face
+            font_style: Str, style of the font face (Thin, Bold, Regular...)
+            antialias: Str, one of "Default", "Best", "Fast", "Good", "None"
+            out_dir: Str, relative path the output directory
+            grayscale: Bool, whether to output grayscale or rgb image
+
+        Raises:
+            (In setters)
+            ValueError: if font_size <= 0 or image_size <= 0
+            ValueError: if font_size < image_size
+            ValueError: if specified font_name cannot be found
+        """
+        # Args: font_size, image_size, out_dir
+        self.image_size = image_size
```

Comment:
Ah it might be confusing here. In Python, you can get the attributes and methods of an object by calling dir(object). Any non-private attribute returned by dir(object) is telling the user that it can be changed. The only way to check validity when an attribute is set is using setter and getters. Here it is actually calling the setter so that the image_size is a positive integer. By doing this, we allow users to change configuration of vg on the go and check for validity in the same  time.